frontend/simple/js/event-log.js

```diff
@@ -0,0 +1,91 @@
+import localforage from 'localforage'
+import {toHash, makeEntry, makeEvent, makeLog} from '../../../shared/functions'
+import request from 'superagent'
+import {EVENTS} from '../../../shared/events'
+import {Log, Group} from '../../../shared/types'
+const {CREATION} = EVENTS
+// =======================
+// Helpers
+// =======================
+let _db = new Map()
+function getDB (hash: string) {
+  if (_db.has(hash)) {
+    return _db.get(hash)
+  }
+  let db = localforage.createInstance({
+    name: 'Group Income',
+    storeName: hash
+  })
+  _db.set(hash, db)
+  return db
+}
+// =======================
+// Exports
+// =======================
+// getItemFromLog retrieves from the log
+// opts object allows for parent hash to be returned instead of data
+export default {
+  async getItemFromLog (log: Log, hash: string) {
+    let db = getDB(log.groupId)
+    let entry = await db.getItem(hash)
+    return entry
+  },
+  // addItemToLog appends to the log
+  // returns the current state of the log
+  async addItemToLog (log: Log, value: any, update ? : boolean) : Log {
```

Comment:
OK, I'm starting to think that passing around this `log` object defeats the purpose of either localForage, or vuex, or both.

Note:

1. `log.groupId` is accessible via `state.currentGroupLog`, so therefore `log.groupId` is extraneous and not needed.
2. `log.currentLogPosition` is also not needed because that's always retrievable via `await db.getItem('currentLogPosition')`

And those are the only two things that are in `log`. In other words, it seems the code is currently keeping track of the same data in multiple places unnecessarily, and I think therefore it can be simplified to remove the need to pass around this `log` object.

One other thing, for the Flow types that are optional, could you please follow the convention that's established in `shared/types.js` and leave no space to indicate optional parameters, e.g.:

```js
async addItemToLog (value: JSONType, update?: boolean) : Log {
```